toolbox/oncall_alert/main.go

```diff
@@ -146,24 +110,63 @@ func main() {
 				log.Printf("No new tests failed in last circle.")
 			}
 		}
-
-		if *autoBlocking {
-			// If any test is in failed status, trying to block, else trying to unblock
-			if failedCount > 0 {
-				blockPRs()
+		if *guardProtectedBranch {
+			if len(newFailures) > 0 {
+				u.BlockMergingOnBranch(githubClnt, *protectedRepo, *protectedBranch)
 			} else {
-				unBlockPRs()
+				u.UnBlockMergingOnBranch(githubClnt, *protectedRepo, *protectedBranch)
 			}
 		}
+		log.Printf("Sleeping for %d seconds", *interval)
+		time.Sleep(time.Duration(*interval) * time.Second)
+	}
+}
 
-		log.Printf("Sleeping for %d minutes", *interval)
-		time.Sleep(time.Duration(*interval) * time.Minute)
+func rerun(job *jobStatus, runNo int) error {
+	cfg, err := getProwJobConfig(job, runNo)
+	if err != nil {
+		return err
+	}
+	if err = triggerConcurrentReruns(job, cfg); err != nil {
+		return err
 	}
+	return nil
 }
 
-func formatMessage(failures map[*postSubmitJob]bool) (mess string) {
-	for job := range failures {
-		mess += fmt.Sprintf("%s failed: %s/%s/%d\n\n", job.name, gubernatorURL, job.name, job.latestRun)
+// Only log this error and continue. Should never exit process.
+func triggerConcurrentReruns(job *jobStatus, cfg ProwJobConfig) error {
+	log.Printf("Rerunning %s\n", job.name)
+	recess := 1 * time.Minute
+	maxRetry := 3
+	for i := 0; i < *numRerun; i++ {
+		if err := u.Retry(recess, maxRetry, func() error {
+			_, e := u.Shell(
+				"kubectl create -f \"https://prow.istio.io/rerun?prowjob=%s\"", cfg.Node)
+			return e
+		}); err != nil {
+			log.Printf("Unable to trigger the %d-th rerun of job %v", i, job.name)
+		}
+	}
+	return nil
+}
+
+func getProwJobConfig(job *jobStatus, runNo int) (ProwJobConfig, error) {
+	cfg := ProwJobConfig{}
```

Comment:
var cfg ProwJobConfig since it is going to be used by jason marshall, that way you can check if it remains nil

Reply:
Done